Edgeai 177 2

matt-auzone opened the pull request:
Updated example files with the latest updates to the schema and validated functionality.

matt-auzone committed:
Modified example files to work with current rt schemas. Examples all tested and verified

matt-auzone committed:
Modified samples to use the python schemas installable. Added pip install command to README

matt-auzone committed:
modified code comments to reference time.sleep() instead of signal.pause() as signal.pause() does not exist on Windows

sebastient commented on README.md:
```diff
@@ -2,6 +2,9 @@
 
 This project demonstrates subscribing to a Maivin EdgeFirst Service from Python and reading messages from the published topics.  The services on Maivin follow a ROS2-like pattern though without using ROS2 itself.  Instead messages are communicated using the [Zenoh](https://zenoh.io) messaging framework with message contents encoded using [CDR](https://en.wikipedia.org/wiki/Common_Data_Representat ...
 
+Please install the Python schemas to run the samples
```
TODO: reminder to update this comment once we've released the edgefirst-schemas update.

sebastient commented on camerainfo.py:
```diff
@@ -2,15 +2,29 @@
 import signal
```
Remove `import signal` if we're not using it anymore.

sebastient commented on detect.py:
```diff
@@ -0,0 +1,62 @@
+import atexit
+import signal
```
remove `import signal`

sebastient commented on camerainfo.py:
```diff
@@ -21,13 +35,12 @@ def _on_exit():
+    # thread to keep the program running.  We use time.sleep() to do this
     # but an application could have its main control loop here instead.
-    signal.pause()
+    time.sleep(args.time)
```
This doesn't adequately replace the previous signal handler.  The idea is to block here until control-c not to add timer and additional parameter.

If signal is not portable and Python doesn't provide better than sleep then we should have a sleep loop.

```
while True:
    time.sleep(1)
```

sebastient commented on detect.py:
```diff
@@ -0,0 +1,62 @@
+    # The declare_subscriber runs asynchronously, so we need to block the main
+    # thread to keep the program running.  We use time.sleep() to do this
+    # but an application could have its main control loop here instead.
+    time.sleep(args.time)
```
Refer to other explanation as to why we need a while True sleep here.

sebastient commented on imu.py:
```diff
@@ -0,0 +1,52 @@
+import atexit
+import signal
```
Remove `import signal`

sebastient commented on imu.py:
```diff
@@ -0,0 +1,52 @@
+    # The declare_subscriber runs asynchronously, so we need to block the main
+    # thread to keep the program running.  We use time.sleep() to do this
+    # but an application could have its main control loop here instead.
+    time.sleep(args.time)
```
As noted above loop on sleep forever.  Remove sleep parameter.

sebastient commented on navsat.py:
```diff
@@ -2,14 +2,43 @@
 import signal
```
Remove signal.

sebastient commented on navsat.py:
```diff
@@ -19,12 +48,12 @@ def _on_exit():
+    # thread to keep the program running.  We use time.sleep() to do this
     # but an application could have its main control loop here instead.
-    signal.pause()
+    time.sleep(args.time)
```
Sleep forever.  Remove parameter.

matt-auzone committed:
Moved everything to infinite loops that will cancel on ctrl-c

matt-auzone committed:
removed additional imports of signal

matt-auzone committed:
added image annotations example

sebastient committed:
Update README.md

Updated pip install documentation to use the released edgefirst-schemas from PyPi.
